includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -774,10 +795,8 @@ public function process_redirect_payment( $order_id, $intent_id, $save_payment_m
 
 				$this->order_service->attach_intent_info_to_order( $order, $intent_id, $status, $payment_method_id, $customer_id, $charge_id, $currency );
 				$this->attach_exchange_info_to_order( $order, $charge_id );
-				$this->update_order_status_from_intent( $order, $intent_id, $status, $charge_id );
 				$this->set_payment_method_title_for_order( $order, $payment_method_type, $payment_method_details );
-
-				self::remove_upe_payment_intent_from_session();
```

Comment:
I also don't think we should delete `remove_upe_payment_intent_from_session` in this method `process_redirect_payment`, as this method is triggered when shoppers/buyers are redirected to the thank-you page. 

At this stage, we should clear up the payment intent session. Otherwise, it will be used for the shopper's next order.

Reply:
Note `$this->update_order_status_from_intent( $order, $intent_id, $status, $charge_id );` call be fore the removed one.
It will trigger an order status update, and as we listen to the hook for order status change, cached intent will be cleared.